File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/playback/service/PlaybackVisualizer.java
package app.zxtune.playback.service;

import app.zxtune.core.Player;
import app.zxtune.playback.Visualizer;

class PlaybackVisualizer implements Visualizer {

  private final Player player;

  PlaybackVisualizer(Player player) {
    this.player = player;
  }

  @Override
  public int getSpectrum(byte[] levels) throws Exception {
    return player.analyze(levels);
  }
}

File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/playback/service/PlaybackVisualizer.kt
package app.zxtune.playback.service

import app.zxtune.core.Player
import app.zxtune.playback.Visualizer

internal class PlaybackVisualizer(private val player: Player) : Visualizer {
    override fun getSpectrum(levels: ByteArray) = player.analyze(levels)
}
